src/utils/clmfates_interfaceMod.F90

```diff
@@ -2593,6 +2643,116 @@ end subroutine wrap_canopy_radiation
+   use FatesDispersalMod,      only : lneighbors, neighbor_type, dispersal_type
+   use FatesInterfaceTypesMod, only : numpft_fates => numpft
+
+   ! Arguments
+   class(hlm_fates_interface_type), intent(inout) :: this
+   logical, optional                              :: is_restart_flag 
+
+   ! Local
+   integer :: numg ! total number of gridcells across all processors
+   integer :: ier  ! error code
+   integer :: g    ! gridcell index
+
+   logical :: set_flag ! local logical variable to pass to IsItDispersalTime
+
+   type (neighbor_type),  pointer :: neighbor
+
+   ! If WrapSeedGlobal is being called at the end a fates restart call,
+   ! pass .false. to the set_dispersed_flag to avoid updating the 
+   ! global dispersal date
+   set_flag = .true.
+   if (present(is_restart_flag)) then
+      if (is_restart_flag) set_flag = .false.   
+   end if
+
+   call t_startf('fates-seed-mpi_reduce')
+
+   if (IsItDispersalTime(setdispersedflag=set_flag)) then
+
+      ! Re-initialize incoming seed buffer for this time step
+      this%fates_seed%incoming_global(:,:) = 0._r8
+      this%fates_seed%outgoing_global(:,:) = 0._r8
+
+      ! Distribute outgoing seed data from all nodes to all nodes
+      call MPI_Allgatherv(this%fates_seed%outgoing_local, procinfo%ncells*numpft_fates, MPI_REAL8, &
+                          this%fates_seed%outgoing_global, this%fates_seed%ncells_array*numpft_fates, this%fates_seed%begg_array*numpft_fates, &
+                          MPI_REAL8, mpicom, ier)
+      write(iulog,*) 'WSG: MPI_Allgatherv ier: ', ier
+
+      ! zero outgoing local for all gridcells outside threaded region now that we've passed them out
+      this%fates_seed%outgoing_local(:,:) = 0._r8
+
+      ! Calculate the current gridcell incoming seed for each gridcell index
+      ! This should be conducted outside of a threaded region to provide access to
+      ! the neighbor%gindex which might not be available in via the clumped index
+      call get_proc_global(ng=numg)
+      do g = 1, numg
+         neighbor => lneighbors(g)%first_neighbor
+         do while (associated(neighbor))
+            ! This also applies the same neighborhood distribution scheme to all pfts
+            ! This needs to have a per pft density probability value
+            this%fates_seed%incoming_global(:,g) = this%fates_seed%incoming_global(:,g) + &
+                                                 this%fates_seed%outgoing_global(:,neighbor%gindex) * &
+                                                 neighbor%density_prob(:) / lneighbors(g)%neighbor_count
+            neighbor => neighbor%next_neighbor
+         end do
+      end do
+
+   endif
+   call t_stopf('fates-seed-mpi_reduce')
+
+ end subroutine WrapSeedGlobal
+
+ ! ======================================================================================
+
+ subroutine wrap_seed_dispersal(this,bounds_clump)
+
+    ! This subroutine pass seed_id_global to bc_in and reset seed_out
```

Comment:
A little hard to understand the comment.

Reply:
I'll update this to remove the out-of-date variable names